dashboard/app/controllers/sessions_controller.rb

```diff
@@ -6,6 +6,10 @@ class SessionsController < Devise::SessionsController
   def new
     session[:return_to] ||= params[:return_to]
     @already_hoc_registered = params[:already_hoc_registered]
+    if params[:providerNotLinked]
+      flash.now[:alert] = I18n.t 'auth.not_linked', provider: I18n.t("auth.#{params[:providerNotLinked]}")
+      @email = params[:email]
```

Comment:
Yes, as you noted above, this is populated in this special case by User.from_omniauth. I'll add a comment here, though, so that it's clear that in general this is not available.